tests/splines/spline_builder.cpp

```diff
@@ -0,0 +1,139 @@
+
+struct IDimX : ddc::NonUniformPointSampling<DimX>
+{
+};
+
+using execution_space = Kokkos::DefaultHostExecutionSpace;
+using memory_space = Kokkos::HostSpace;
+
+TEST(SplineBuilder, ShortInterpolationGrid)
+{
+    CoordX const x0(0.);
+    CoordX const xN(1.);
+    std::size_t const ncells = 5;
+
+    ddc::init_discrete_space<BSplinesX>(x0, xN, ncells);
+
+    // One point missing
+    std::vector<double> const range {0.1, 0.3, 0.5, 0.7};
+
+    ddc::DiscreteDomain<IDimX> const interpolation_domain
+            = ddc::init_discrete_space<IDimX>(IDimX::init<IDimX>(range));
+
+    EXPECT_THROW(
+            (ddc::SplineBuilder<
+                    execution_space,
+                    memory_space,
+                    BSplinesX,
+                    IDimX,
+                    ddc::BoundCond::PERIODIC,
+                    ddc::BoundCond::PERIODIC,
+                    ddc::SplineSolver::GINKGO,
+                    IDimX>(interpolation_domain)),
+            std::runtime_error);
+}
+
+TEST(SplineBuilder, LongInterpolationGrid)
+{
+    CoordX const x0(0.);
+    CoordX const xN(1.);
+    std::size_t const ncells = 5;
+
+    ddc::init_discrete_space<BSplinesX>(x0, xN, ncells);
+
+    // One point too much
+    std::vector<double> const range {0.1, 0.3, 0.5, 0.7, 0.9, 0.95};
+
+    ddc::DiscreteDomain<IDimX> const interpolation_domain
+            = ddc::init_discrete_space<IDimX>(IDimX::init<IDimX>(range));
+
+    EXPECT_THROW(
+            (ddc::SplineBuilder<
+                    execution_space,
+                    memory_space,
+                    BSplinesX,
+                    IDimX,
+                    ddc::BoundCond::PERIODIC,
+                    ddc::BoundCond::PERIODIC,
+                    ddc::SplineSolver::GINKGO,
+                    IDimX>(interpolation_domain)),
+            std::runtime_error);
+}
+
+TEST(SplineBuilder, BadShapeInterpolationGrid)
+{
+    CoordX const x0(0.);
+    CoordX const xN(1.);
+    std::size_t const ncells = 5;
+
+    ddc::init_discrete_space<BSplinesX>(x0, xN, ncells);
+
+    // All points end up in the first cell ]0, 0.2[
+    std::vector<double> const range {0.1, 0.11, 0.12, 0.13, 0.14};
+
+    ddc::DiscreteDomain<IDimX> const interpolation_domain
+            = ddc::init_discrete_space<IDimX>(IDimX::init<IDimX>(range));
+
+    EXPECT_THROW(
+            (ddc::SplineBuilder<
+                    execution_space,
+                    memory_space,
+                    BSplinesX,
+                    IDimX,
+                    ddc::BoundCond::PERIODIC,
+                    ddc::BoundCond::PERIODIC,
+                    ddc::SplineSolver::GINKGO,
+                    IDimX>(interpolation_domain)),
+            std::runtime_error);
+}
+
+TEST(SplineBuilder, CorrectInterpolationGrid)
+{
+    CoordX const x0(0.);
+    CoordX const xN(1.);
+    std::size_t const ncells = 5;
+
+    ddc::init_discrete_space<BSplinesX>(x0, xN, ncells);
+
+    std::vector<double> const range {0.1, 0.11, 0.25, 0.23, 0.24};
```

Comment:
These are not ordered. Is that intentional? Should we also check this? (The points should be ordered)

Reply:
good catch, and we should detect that! What do you think in the construction of the NonUniformPointSampling ? (you do not have unordered tricks in Gysela ?)

Reply:
I think the construction of the NonUniformPointSampling seems like a logical place for this and that will be more general than putting it in `check_n_points_in_cell`

We don't use unordered points in Gysela

Reply:
Fixed in 04523404841cbe4ac863839a5be1797d41bd7a55, the sorting check is already in `main`